enos/modules/build_boundary_docker_local/build.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+set -eux -o pipefail
+
+env
+
+# Go to the root of the boundary repo
+root_dir="$(git rev-parse --show-toplevel)"
+pushd "${root_dir}" > /dev/null
+
+docker buildx inspect multiarch || docker buildx create --use --driver docker-container --name multiarch
+docker buildx build \
+    --load \
+    --platform ${GOOS}/${GOARCH} \
+    --tag ${IMAGE_NAME} \
+    --target=dev \
+    --build-arg=boundary \
+    --output="type=docker,dest=${ARTIFACT_PATH}" \
+    .
+docker buildx rm multiarch
```

Comment:
Should this be added to the Makefile? cc: @tmessi

Reply:
I know we have the [docker-build-dev](https://github.com/hashicorp/boundary/blob/main/Makefile#L358-L368) target already. Might be able to update that to support passing through additional flags to `docker buildx`

Reply:
👍 I think I see some refactoring I can do with that makefile target. Giving it a shot